src/Catty/DataModel/Bricks/Sound/SetTempoToBrick.swift

```diff
@@ -0,0 +1,90 @@
+/**
+ *  Copyright (C) 2010-2021 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc(SetTempoToBrick)
+@objcMembers class SetTempoToBrick: Brick, BrickFormulaProtocol {
+
+    private var tempo: Formula
+
+    override required init() {
+        self.tempo = AudioEngineDefines.defaultTempo!
+        super.init()
+        let value = Int(self.tempo.formulaTree.value)!
+        if value < 20 { self.tempo = Formula(integer: 20) }
+        if value > 500 { self.tempo = Formula(integer: 500) }
+    }
+
+    func category() -> kBrickCategoryType {
+        kBrickCategoryType.soundBrick
+    }
+
+    override class func description() -> String {
+        "SetTempoToBrick"
+    }
+
+    override func getRequiredResources() -> Int {
+        tempo.getRequiredResources()
+
+    }
+
+    override func brickCell() -> BrickCellProtocol.Type! {
+        SetTempoToBrickCell.self as BrickCellProtocol.Type
+    }
+
+    func formula(forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) -> Formula! {
+        self.tempo
+    }
+
+    func setFormula(_ formula: Formula!, forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) {
+        let value = Int(formula.formulaTree.value!)!
+        if value < 20 { self.tempo = Formula(integer: 20) } else if value > 500 { self.tempo = Formula(integer: 500) } else { self.tempo = formula }
```

Comment:
This validation should not be done here since the value of a formula is variable, i.e., when using a sensor it is different each time it is evaluated. This also affects the initializer.